UpdatedImageProcessing/TargetDetection/color_operations.py

```diff
@@ -0,0 +1,120 @@
+import cv2
+import math
+import numpy
+import PIL.ImageOps
+from PIL import Image
+from sklearn.cluster import MiniBatchKMeans
+
+class ColorOperations(object):
+
+    @staticmethod
+    def find_distance(color_1, color_2):
+        return math.sqrt(((color_1[0] - color_2[0]) ** 2) + ((color_1[1] - color_2[1]) ** 2) + ((color_1[2] - color_2[2]) ** 2))
```

Comment:
You can use numpy here to create a vectorized operation that works in an n-dimensional space. Please adapt it to use that (which will also be faster).